File: 1.two-sum.py
#
# @lc app=leetcode id=1 lang=python3
#
# [1] Two Sum
#

# @lc code=start
from typing import List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(0, len(nums)):
            for j in range(i+1, len(nums)):
                if(target == nums[i] + nums[j]):
                    return [i, j]
# @lc code=end

print(Solution().twoSum([2,7,11,15], 9))

File: blind75/1.two-sum.java
import java.util.HashMap;

/*
 * @lc app=leetcode id=1 lang=java
 *
 * [1] Two Sum
 */

// @lc code=start
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer, Integer> mapping = new HashMap<>();
        for(int i=0; i<nums.length; i++) {
            if(mapping.containsKey(nums[i])){
                return new int[] {i, mapping.get(nums[i])};
            }
            mapping.put(target - nums[i], i);
        }
        return new int[] {};
    }
}
// @lc code=end

